### ISSUE TITLE
$THREAD_SUBPROCS=False freezes pipes with aliases

### ISSUE THREAD

**Issue by anki-code** [xonsh/xonsh#3804]
Hi! xonsh is the best of the best! Thanks!

The `xontrib-pipeliner` user found that `bashisms` kills pipeliner (https://github.com/anki-code/xontrib-pipeliner/issues/1). I deep dive and found that @scopatz added `$THREAD_SUBPROCS=False` to bashisms (https://github.com/xonsh/xonsh/commit/dc8728bdafcf3c1967305df57451c225e8bf8ba2) and this is the cause.

How to repeat:
```python
$THREAD_SUBPROCS=False
def _pl(args, stdin, stdout):
    print(1)

aliases['pl'] = _pl

echo 1 | pl

# XONSH_DEBUG=2 will tell:
#<xonsh-code>:7:5 - echo 1 | pl
#<xonsh-code>:7:5 + ![echo 1 | pl]
# And you'll have freezed xonsh and 100% CPU usage.
```

Please fix or advice how to fix it for bashisms and xontrib-pipeliner.

Thanks!

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by scopatz** [comment]
Yeah, probably the easiest fix is for xontrib-pipeliner to set `$THREAD_SUBPROCS=True`

**Comment by anki-code** [comment]
Because we're talking about global variable `$THREAD_SUBPROCS` I don't know why this is a good way. I do `xontrib load bashisms` and the variable going to `False` then I run `xontrib load pipeliner` and the variable going to `True` and bashisms will not work? I'm going to leave this broken because I see many issues around THREAD_SUBPROC and the mess around it.

**Comment by scopatz** [comment]
Yeah, this is one of those things that it hard to know what the right strategy is. The proper fix is to rewrite the whole subproc execution model.

**Comment by anki-code** [comment]
I pray every evening that xore is working on it. 🙏

**Comment by scopatz** [comment]
haha I am really trying to find time to do work on it.

**Comment by anki-code** [comment]
Closed with reference to #4250
